### PR TITLE
Exposes props of the selection check button in the GroupedList to enable aria-label

### LINKED ISSUE #7751: DetailsList GroupHeader checkbox is missing aria-label
This is the same issue as #5986 which was closed with a different solution. I think we may still want the aria-label.

### Environment Information 
 
- __Package version(s)__: 6.129.1
- __Browser and OS versions__: Chrome (Windows 10) Version 71.0.3578.98 (Official Build) (64-bit)

### Please provide a reproduction of the bug in a codepen:

Reusing this [codepen ](
https://codepen.io/peterliumich/pen/jvNWRy?editors=1111)from #5986 
 
#### Actual behavior:
 
The checkbox-button does not have an aria-label.
 
#### Expected behavior:

The checkbox-button has an aria-label. 

In the linked PR [discussion](https://github.com/OfficeDev/office-ui-fabric-react/pull/5987) the following examples are linked which don't use aria-labels in this case:

> Here are the examples I was looking at:
> - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_checkbox_role
> - https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/checkbox/checkbox-2.html

Guru (@iamrafan) mentioned that the linked examples are a bit misleading because the checkbox has existing content a screen reader can use. In our OfficeFabric case, the checkbox has no content so an aria-label would provide some helpful context.

Maybe something like `expandButtonProps` here would be helpful for the checkbox:
https://github.com/OfficeDev/office-ui-fabric-react/blob/59276033edb69b6e845dde243fa3e9fce2bc0f19/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx#L21

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No) see PR [here](https://github.com/OfficeDev/office-ui-fabric-react/pull/7777)
Requested priority: (Blocking, High, Normal, Low)  normal

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #7751
- [x] Include a change request file using `$ npm run change`

~~Would appreciate help here - I'm getting: `ERROR: Command failed: git diff upstream/master... --dirstat=files,0
fatal: ambiguous argument 'upstream/master...': unknown revision or path not in the working tree.
`~~ (Needed to explicitly fetch upstream)
#### Description of changes

This exposes props of the selection check in the group header component so we can add an aria-label. It mirrors the similar change made by Natalie [here](https://github.com/OfficeDev/office-ui-fabric-react/pull/5249).

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
Thanks for raising this PR @karanbirsingh and for performing all the steps to get a passing build 👏! This change is a good one and interesting from an API perspective. 

Initially, I thought we might want to collapse the props `expandButtonProps` and `selectionCheckProps` into one. That was because both props _only_ Object spread native props on a `<button/>`. However, I can see the need for providing _different_ values for attributes such as `aria-label` depending on the use-case, select OR expand. If we consolidated props, we'd have to recommend developers handle the conditional logical themselves. That seems like it might be burdensome as a user of the library.

What are your thoughts on the above? I'd like to run your pull-request this by @natalieethell and @aditima tomorrow prior to signing-off.

**Comment by karanbirsingh** [comment]
>However, I can see the need for providing _different_ values for attributes such as `aria-label` depending on the use-case, select OR expand. If we consolidated props, we'd have to recommend developers handle the conditional logical themselves. That seems like it might be burdensome as a user of the library.

Hi Kevin, can you provide an example of what this conditional logic might look like? For example, if I wanted slightly different aria-labels for the checkbox and button, I'm not sure how I could do so if there were a single `props` object.

**Comment by KevinTCoughlin** [comment]
> 
> 
> > However, I can see the need for providing _different_ values for attributes such as `aria-label` depending on the use-case, select OR expand. If we consolidated props, we'd have to recommend developers handle the conditional logical themselves. That seems like it might be burdensome as a user of the library.
> 
> Hi @KevinTCoughlin Kevin Coughlin FTE - can you provide an example of what this conditional logic might look like? For example, if I wanted slightly different aria-labels for the checkbox and button, I'm not sure how I could do so if there were a single `props` object.

I spoke with @natalieethell today and we agreed that collapsing the props would not be ideal for consumers of the library.

However, should this prop `expandButtonProps` be added to `IGroupHeaderProps` rather than `IGroupDividerProps`?

There's currently a PR out by @Vitalius1 which addresses our misuse of prop types in these components, #7776.

@Vitalius1, given your recent work with this component what are your thoughts on the above re: `IGroupHeaderProps` instead of `IGroupDividerProps`?

Is there a case where we would use these props while rendering a footer? They seem specific to the default header case.

**Comment by Vitalius1** [comment]
I do think it's a Header specific case and don't see there is a need to have it under the shared `IGroupDividerProps`. I am only worried about the discussion in #5249 where we had some reserves in having English words as a default to a prop like this.

**Comment by karanbirsingh** [comment]
>However, should this prop expandButtonProps be added to IGroupHeaderProps rather than IGroupDividerProps?

@KevinTCoughlin do you mean `selectionCheckProps` (or both)? Given the related change by @Vitalius1, let me know how we should proceed with this PR.

**Comment by KevinTCoughlin** [comment]
> 
> 
> > However, should this prop expandButtonProps be added to IGroupHeaderProps rather than IGroupDividerProps?
> 
> @KevinTCoughlin Kevin Coughlin FTE do you mean `selectionCheckProps` (or both)? Given the related change by @Vitalius1 Vitalie Braga (Webtellect LLC) Vendor, let me know how we should proceed with this PR.

@karanbirsingh, let's wait for @Vitalius1's changes to be merged. Then I would suggest modifying your change-set to something like the following:

```diff
diff --git a/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx b/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx
index 66b591c0d..f3bf25c2d 100644
--- a/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx
+++ b/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx
@@ -57,6 +57,7 @@ export class GroupHeaderBase extends BaseComponent<IGroupHeaderProps, IGroupHead
       onRenderTitle = this._onRenderTitle,
       isCollapsedGroupSelectVisible = true,
       expandButtonProps,
+      selectAllButtonProps,
       theme,
       styles,
       className,
@@ -100,6 +101,7 @@ export class GroupHeaderBase extends BaseComponent<IGroupHeaderProps, IGroupHead
               aria-checked={!!currentlySelected}
               data-selection-toggle={true}
               onClick={this._onToggleSelectGroupClick}
+              {...selectAllButtonProps}
             >
               <Check checked={currentlySelected} />
             </button>
diff --git a/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.types.ts b/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.types.ts
index c5b5ba587..902196658 100644
--- a/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.types.ts
+++ b/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.types.ts
@@ -22,6 +22,16 @@ export interface IGroupHeaderProps extends IGroupDividerProps {
    * GroupedList id for aria-controls
    */
   groupedListId?: string;
+
+  /**
+   * Native props for the GroupHeader expand and collapse button
+   */
+  expandButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
+
+  /**
+   * Native props for the GroupHeader select all button
+   */
+  selectAllButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
 }
 
 export type IGroupHeaderStyleProps = Required<Pick<IGroupHeaderProps, 'theme'>> &
diff --git a/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.types.ts b/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.types.ts
index 14d79984a..3c8cb370f 100644
--- a/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.types.ts
+++ b/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.types.ts
@@ -295,7 +295,10 @@ export interface IGroupDividerProps {
   /** Override which allows the caller to provider a custom title. */
   onRenderTitle?: IRenderFunction<IGroupHeaderProps>;
 
-  /** Props for expand/collapse button */
+  /**
+   * Props for expand/collapse button
+   * @deprecated Use {@link IGroupHeaderProps.expandButtonProps} instead.
+   */
   expandButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
 
   /** Stores parent group's children. */
```

I would avoid setting default string literal text for the reasons laid out in #5249. You _should_ still be able to provide that string for your use-case, but should confirm prior to merge. We just want to avoid it being the default case.

**Comment by karanbirsingh** [comment]
@KevinTCoughlin can you please take a look at the updated changes? The props have been moved to GroupHeaderProps and the old expandButtonProp has been deprecated.

File: common/changes/office-ui-fabric-react/groupheader-aria_2019-01-24-00-29.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Exposes props of selection check button in group header",
```
Comment (natalieethell) [inline common/changes/office-ui-fabric-react/groupheader-aria_2019-01-24-00-29.json]: My one small suggestion would be to add the affected component to the start of the comment, perhaps something like "GroupHeader: Exposes props of selection check button"

Comment (Vitalius1) [inline common/changes/office-ui-fabric-react/groupheader-aria_2019-01-24-00-29.json]: @natalieethell we can do suggestions in our comments 😄 
```suggestion
      "comment": "GroupHeader: exposes props of selection check button.",
```

Comment (natalieethell) [reply common/changes/office-ui-fabric-react/groupheader-aria_2019-01-24-00-29.json]: Oh yeah I keep forgetting! Nice!

Comment (karanbirsingh) [reply common/changes/office-ui-fabric-react/groupheader-aria_2019-01-24-00-29.json]: I've committed the suggestion - thanks!

**Pull request merged**